dcicutils/license_utils.py

```diff
@@ -762,648 +780,230 @@ def __init__(self, message=None, unacceptable_licenses=None):
-                # Ref: https://redresscompliance.com/mysql-license-a-complete-guide-to-licensing/#:~:text=commercial%20use  # noQA
-                'mysql-common',
-                'mysql-libs',
-
-                # This is our own library
-                'r-scan2', 'scan2',
-            ]
-        }
-    )
 
-    EXPECTED_MISSING_LICENSES = ParkLabGplPipelineLicenseChecker.EXPECTED_MISSING_LICENSES + [
+def merge_policy_strings_or_regexps(*, policy_name, policy_datum, other_policy_data):
+    return merge_policy_lists(policy_name=policy_name, policy_datum=policy_datum, other_policy_data=other_policy_data,
+                              sort_key=str_or_regexp_sort_key)
+
+
+def merge_policy_dicts(*, policy_name, policy_datum, other_policy_data):
+    ignored(policy_name)
+    merged = defaultdict(lambda: [])
+
+    def add_to_merged(d):
+        for k, values in d.items():
+            for value in values:
+                merged[k].append(value)
+
+    add_to_merged(policy_datum)
+    for other_datum in other_policy_data:
+        add_to_merged(other_datum)
+
+    return {k: sorted(set(v)) for k, v in sorted(merged.items())}
+
+
+POLICY_ATTRS: callable = {
+    'class_key': use_policy_literal,
+    'class_name': use_policy_literal,
+    'inherits_from': use_policy_literal,
+    'description': use_policy_literal,
+    'LICENSE_TITLE': use_policy_literal,
+    'COPYRIGHT_OWNER': use_policy_literal,
+    'LICENSE_FRAMEWORKS': use_policy_literal,
+    'ALLOWED': merge_policy_strings_or_regexps,
+    'EXPECTED_MISSING_LICENSES': merge_policy_lists,
+    'EXCEPTIONS': merge_policy_dicts,
+}
+
+POLICY_MERGE_LISTS = {'ALLOWED', 'EXPECTED_MISSING_LICENSES'}
+POLICY_MERGE_DICTS = {'EXCEPTIONS'}
+
+
+def get_attrs_for_classes(attr: str, class_data: List[Type]):
+    result = []
+    for class_datum in class_data:
+        attr_val = getattr(class_datum, attr, None)  # Intentionally treats explicit None the same as missing
+        if attr_val is not None:
+            result.append(attr_val)
+    return result
+
+
+def default_policy_data(*, policy_name: str, policy_data: AnyJsonData, parent_classes: List[Type]):
+    result = {}
+    for key_to_default, val_to_be_defaulted in policy_data.items():
+        attr_handler: Optional[callable] = POLICY_ATTRS.get(key_to_default)
+        if attr_handler is None:
+            raise ValueError(f"Bad policy attribute: {key_to_default}")
+        result[key_to_default] = attr_handler(policy_name=policy_name, policy_datum=val_to_be_defaulted,
+                                              other_policy_data=get_attrs_for_classes(key_to_default, parent_classes))
+    return result
+
+
+def load_license_policies(for_env, policy_dir=None):
+    for policy_name in built_in_policy_names():
+        find_or_create_license_class(policy_name=policy_name, policy_dir=policy_dir, for_env=for_env)
+
 
-    ]
+load_license_policies(for_env=globals())
```

Comment:
Import side effect desired here?

Reply:
Well, because of the naming conventions, it can't collide adversely with other things in the environment. It's going to suffix "LicenseChecker", so it can only end up in a namespace of license checkers, and the decorator for the name will not let you redefine a license checker.

The import side effect here is that dcicutils.license_utils will export the names of the pre-defined policies. (In testing, I have it export to locals() so I can test that it puts them into the designated env. But I've interactively tested that importing this library does get the same classes that were in the original PR.